source/fundamentals/builders/projections.txt

```diff
@@ -11,3 +11,169 @@ Projections Builders
+
+.. code-block:: java
+
+   excludeId()
+
+.. note::
+
+   ``excludeId`` is a convenience method and is shorthand for:
+
+   .. code-block:: java
+
+      exclude("_id")
+
+Project an Array Element Match
+------------------------------
+
+Use the ``elemMatch`` methods to specify an array projection.
+
+The ``elemMatch(String, Bson)`` method variant is used to specify an array projection that will include the first
+element of an array that matches a supplied query filter.
+
+.. note::
+
+   Only the first element that matches the specified query filter will be included, regardless of how many matches there
+   may be.
+
+The following example projects the first element of the :guilabel:`orders` array where the :guilabel:`quantity` field is
+greater that ``3``:
+
+.. code-block:: java
+
+   elemMatch("orders", Filters.gt("quantity", 3))
+
+
+
+Use the ``elemMatch(String)`` method variant to specify a
+:manual:`positional projection </reference/operator/projection/positional/#projection/>` to include the first element of an array
+that matches the filter supplied as part of the query.
+
+.. warning::
+
+   In MongoDB versions < 4.4, the specified array field must appear in the query document. Beginning in MongoDB 4.4,
+   you can use a positional project on an array field that does not appear in the query document.
+
+
+The following example projects the first element of the :guilabel:`orders` array:
+
+.. code-block:: java
+
+   elemMatch("orders")
+
+.. tip::
+
+   Specifying ``elemMatch(String fieldName, Bson filter)`` in the projection causes MongoDB to use the
+   :manual:`projection $elemMatch operator </reference/operator/projection/elemMatch/>`. To use the
+   :manual:`$elemMatch query operator </reference/operator/query/elemMatch/>` as part of your search criteria, see the
+   :ref:`Filters builders page <filters-builders>`.
+
+Slice
+-----
+
+Use the ``slice`` methods to project a :manual:`slice </reference/operator/projection/slice/>` of an array.
+
+The following example projects the first **7** elements of the :guilabel:`tags` array:
+
+.. code-block:: java
+
+  slice("tags", 7)
+
+The following example skips the first **2** elements of the :guilabel:`tags` array and projects the next **5**:
+
+.. code-block:: java
+
+   slice("tags", 2, 5)
+
+Text Score
+----------
+
+Use the ``metaTextScore`` method to specify a projection of
+:manual:`score of a text query </reference/operator/query/text/#return-the-text-search-score/>`
+
+The following example projects the text score as the value of the :guilabel:`score` field:
+
+.. code-block:: java
+
+   metaTextScore("score")
+
+Combining Projections
+---------------------
+
+Use the ``fields`` method to combine multiple projections.
+
+The following example includes the :guilabel:`quantity` and :guilabel:`totalAmount` fields and excludes the
+:guilabel:`_id` field:
+
+.. code-block:: java
+
+   fields(include("quantity", "totalAmount"), excludeId())
```

Comment:
I was expecting, at the end of the page, a more complete code example showing how to put this in action. Does that exist elsewhere?

Reply:
There are more complex examples in another PR for indexes (which shows filters, sorts, and projections). Trying to keep them lightweight for now, we can add denser examples should users request it.